signing/mcl/multisig/vectors_test.go

```diff
@@ -0,0 +1,199 @@
+package multisig
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/hashing/blake2b"
+	"github.com/stretchr/testify/require"
+
+	crypto "github.com/multiversx/mx-chain-crypto-go"
+	"github.com/multiversx/mx-chain-crypto-go/signing"
+	"github.com/multiversx/mx-chain-crypto-go/signing/mcl"
+)
+
+// TestVector defines the data structure used to unmarshal the JSON file
+type TestVector struct {
+	testName      string
+	message       []byte
+	publicKeys    []crypto.PublicKey
+	signatures    [][]byte
+	aggregatedSig []byte
+	expectedError error
+}
+
+func TestAggregateSignaturesKOSK(t *testing.T) {
+	t.Parallel()
+
+	lls := &BlsMultiSignerKOSK{}
+
+	err := generateJSONFileKOSKForAggregateSignaturesTests(lls)
+	require.Nil(t, err)
+
+	suite := mcl.NewSuiteBLS12()
+	testVectors, err := createTestSetup(suite, "multisigKOSKAggSig.json")
+	require.Nil(t, err)
+
+	for _, testVector := range testVectors {
+		testName := testVector.testName
+		signatures := testVector.signatures
+		pubKeys := testVector.publicKeys
+		expectedError := testVector.expectedError
+		aggregatedSig := testVector.aggregatedSig
```

Comment:
new vars not necessarily needed

valid for all tests

Reply:
updated